src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService_FindDeclaration.cs

```diff
@@ -164,16 +172,20 @@ class NestedType
                 }
             }
 
-            // Generate into any decl we can find.
+            // If there is a declaration in a non auto-generated file, prefer it.
+            Func<Document,bool> isGeneratedCode = 
+                solution.Workspace.Services.GetService<IGeneratedCodeRecognitionService>().IsGeneratedCode;
```

Comment:
I'm not saying that the user wants to add code in .Designer - I was saying that name of the Func property returned by the IGeneratedCodeRecognitionService is misleading. :)

Ignore the comment - nothing to see here, just making an observation. :)